src/components/vm/nics/nicAdd.tsx

```diff
@@ -62,10 +67,26 @@ function getRandomMac(vms) {
     }
 
     console.warn("Could not generate non-conflicting MAC address");
-    return undefined;
+    return "";
```

Comment:
This gets passed to `domainAttachIface()` as-is, i.e. changes from `undefined` to empty string. That is a code smell, but domainAttachIface() does `const macArg = mac ? "mac=" + mac + "," : "";` so it should be fine.

:+1: